src/main/java/io/pinecone/PineconeIndexOperationClient.java

```diff
@@ -1,52 +1,77 @@
 package io.pinecone;
 
 import io.pinecone.model.CreateIndexRequest;
-import org.asynchttpclient.AsyncHttpClient;
-import org.asynchttpclient.DefaultAsyncHttpClient;
-
+import okhttp3.*;
 import java.io.IOException;
+import static io.pinecone.utils.Constants.*;
 
 public class PineconeIndexOperationClient {
-    private AsyncHttpClient client;
-    private PineconeClientConfig clientConfig;
-    private String url;
+    private final OkHttpClient client;
+    private final PineconeClientConfig clientConfig;
+    private final String url;
 
-    PineconeIndexOperationClient(PineconeClientConfig clientConfig, AsyncHttpClient client) {
-        this.clientConfig = clientConfig;
+    private PineconeIndexOperationClient(PineconeClientConfig clientConfig, OkHttpClient client, String url) {
         this.client = client;
-        this.url = "https://controller." + clientConfig.getEnvironment() + ".pinecone.io/databases/";
+        this.clientConfig = clientConfig;
+        this.url = url;
+    }
+
+    public PineconeIndexOperationClient(PineconeClientConfig clientConfig, OkHttpClient client) {
+        this(clientConfig, client, createUrl(clientConfig));
     }
 
     public PineconeIndexOperationClient(PineconeClientConfig clientConfig) {
-        this.clientConfig = clientConfig;
-        client = new DefaultAsyncHttpClient();
-        this.url = "https://controller." + clientConfig.getEnvironment() + ".pinecone.io/databases/";
+        this(clientConfig, new OkHttpClient());
+    }
+
+    private static String createUrl(PineconeClientConfig clientConfig) {
+        if (clientConfig.getApiKey() == null || clientConfig.getEnvironment() == null) {
+            throw new PineconeValidationException("Both API key and environment name are required for index operations.");
```

Comment:
The validation is required to prevent null pointer exception since if PineconeClientConfig is created without either apikey or environment, then the index operations cannot happen